hail/python/hail/matrixtable.py

```diff
@@ -539,6 +539,136 @@ class MatrixTable(ExprContainer):
     def _from_java(jmir):
         return MatrixTable(ir.JavaMatrix(jmir))
 
+    @staticmethod
+    @typecheck(
+        globals=nullable(dictof(str, anytype)),
+        rows=nullable(dictof(str, sequenceof(anytype))),
+        cols=nullable(dictof(str, sequenceof(anytype))),
+        entries=nullable(dictof(str, sequenceof(sequenceof(anytype)))),
+    )
+    def from_parts(globals=None, rows=None, cols=None, entries=None) -> 'MatrixTable':
```

Comment:
I'm not sure if this is team-wide standard, but I prefer that we include mypy type annotations in new code. I would like us to eventually actually enforce that mypy succeeds on the `hail/python/hail` code. Currently we only require that of the "services" code (ci, batch, etc.).